tensorflow/lite/experimental/examples/unity/TensorFlowLitePlugin/Assets/TensorFlowLite/SDK/Scripts/Interpreter.cs

```diff
@@ -89,18 +94,95 @@ public class Interpreter : IDisposable
           tensor, tensorDataPtr, Buffer.ByteLength(outputTensorData)));
     }
 
+    public string GetInputTensorInfo(int index) {
+      TfLiteTensor tensor = TfLiteInterpreterGetInputTensor(interpreter, index);
+      return GetTensorInfo(tensor);
+    }
+
+    public string GetOutputTensorInfo(int index) {
+      TfLiteTensor tensor = TfLiteInterpreterGetOutputTensor(interpreter, index);
+      return GetTensorInfo(tensor);
+    }
+
+    public static string GetVersion() {
+      return Marshal.PtrToStringAnsi(TfLiteVersion());
+    }
+
+    private static string GetTensorName(TfLiteTensor tensor) {
+      return Marshal.PtrToStringAnsi(TfLiteTensorName(tensor));
+    }
+
+    private static string GetTensorInfo(TfLiteTensor tensor) {
+      var sb = new System.Text.StringBuilder();
+      sb.AppendFormat("{0} type:{1}, dims:[",
+          GetTensorName(tensor),
+          TfLiteTensorType(tensor));
+      
+      int dims = TfLiteTensorNumDims(tensor);
+      for (int i = 0; i < dims; i++) {
+        sb.Append(TfLiteTensorDim(tensor, i));
+        sb.Append(i == dims - 1 ? "]" : ", ");
+      }
+      return sb.ToString();
+    }
+
     private static void ThrowIfError(int resultCode) {
       if (resultCode != 0) throw new Exception("TensorFlowLite operation failed.");
     }
 
     #region Externs
 
+    #if UNITY_IPHONE && !UNITY_EDITOR
+    private const string TensorFlowLibrary = "__Internal";
+#else
+    private const string TensorFlowLibrary = "tensorflowlite_c";
+#endif
+
+    public enum TfLiteType {
+      NoType = 0,
+      Float32 = 1,
+      Int32 = 2,
+      UInt8 = 3,
+      Int64 = 4,
+      String = 5,
+      Bool = 6,
+      Int16 = 7,
+      Complex64 = 8,
+      Int8 = 9,
+      Float16 = 10,
+    }
+
+    public struct TfLiteQuantizationParams {
+      public float scale;
+      public int zero_point;
+    }
+
+    [DllImport (TensorFlowLibrary)]
+    private static extern unsafe IntPtr TfLiteVersion();
+
     [DllImport (TensorFlowLibrary)]
     private static extern unsafe TfLiteInterpreter TfLiteModelCreate(IntPtr model_data, int model_size);
 
     [DllImport (TensorFlowLibrary)]
     private static extern unsafe TfLiteInterpreter TfLiteModelDelete(TfLiteModel model);
 
+    [DllImport (TensorFlowLibrary)]
+    private static extern unsafe TfLiteInterpreterOptions TfLiteInterpreterOptionsCreate();
+
+    [DllImport (TensorFlowLibrary)]
+    private static extern unsafe void TfLiteInterpreterOptionsDelete(TfLiteInterpreterOptions options);
+
+    [DllImport (TensorFlowLibrary)]
+    private static extern unsafe TfLiteInterpreterOptions TfLiteInterpreterOptionsSetNumThreads(
+        TfLiteInterpreterOptions options,
+        int num_threads
+    );
+
+    [DllImport (TensorFlowLibrary)]
```

Comment:
I would say let's leave this out until we have it wired up?

Reply:
ok, I am implementing GPU delegate. I'll add these in the next PR.
https://github.com/asus4/tf-lite-unity-sample